cmd/kubeadm/app/cmd/phases/join/kubelet.go

```diff
@@ -93,13 +93,22 @@ func getKubeletStartJoinData(c workflow.RunData) (*kubeadmapi.JoinConfiguration,
 // runKubeletStartJoinPhase executes the kubelet TLS bootstrap process.
 // This process is executed by the kubelet and completes with the node joining the cluster
 // with a dedicates set of credentials as required by the node authorizer
-func runKubeletStartJoinPhase(c workflow.RunData) error {
+func runKubeletStartJoinPhase(c workflow.RunData) (returnErr error) {
 	cfg, initCfg, tlsBootstrapCfg, err := getKubeletStartJoinData(c)
 	if err != nil {
 		return err
 	}
 	bootstrapKubeConfigFile := kubeadmconstants.GetBootstrapKubeletKubeConfigPath()
 
+	defer func() {
+		// Deletes the bootstrapKubeConfigFile, so the credential used for TLS bootstrap are removed from disk
+		deleteErr := os.Remove(bootstrapKubeConfigFile)
+		// Only return a delete error if we weren't already returning something else
+		if returnErr == nil && deleteErr != nil {
```

Comment:
Um...
I think the deletion should be best-effort.
Even if it fails, it should not fail the entire `KubeletStartJoinPhase`.
Is it better to log this instead of returning an error?

Reply:
i agree.
this should be just:
```
defer os.Remove(bootstrapKubeConfigFile)
```

Reply:
Alrighty, I'll change this out.

Reply:
I would think it would be good to at least log if it fails, being that it is part of the process, but that's just my thought.

Reply:
I don't agree with this.
Deleting this token from disk is a matter of security.

Also, that decision wasn't made in this PR.
This patch is a bugfix for deleting the file in spite of earlier error conditions.

The previously merged patch was already wrapping and returning this error.